.github/workflows/main-publish.yml

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2020 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+name: Main Publish
+
+on:
+  workflow_run:
+    workflows: ["Main CI"]
+    branches: [ main ]
+    types:
+      - completed
+
+jobs:
+  build:
+    name: Publish main
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'projectnessie/nessie' }}
+
+    # Runs in the `snapshot` environment, which has the necessary secrets.
+    # See https://docs.github.com/en/actions/reference/environments
+    environment: snapshot
+
+    steps:
+
+    ### BEGIN runner setup
+    - name: Update /etc/hosts
+      run: echo -e "$(ip addr show eth0 | grep "inet\b" | awk '{print $2}' | cut -d/ -f1)\t$(hostname -f) $(hostname -s)" | sudo tee -a /etc/hosts
```

Comment:
can we just comment on why we have to do this so we don't forget?

Reply:
actually I thikn it can be removed for this workflow as spark isn't being run in tests. Can we add the comment to the test workflow?

Reply:
Not sure whether not applying that workaround would not cause issues as it's a network/machine/config issue. That's why I initially opted to apply it everywhere. But yea, I've removed it elsewhere.